pytest_lab/network.py

```diff
@@ -94,4 +94,10 @@ def inner(count):
 @pytest.fixture(scope='class')
 def vlan_addr(vlan, addr_family):
     '''Return the correct ip address for the macvlan interface'''
-    return vlan.get_address(addr_family).compressed
+    return vlan.get_address(addr_family)[0].compressed
+
+
+@pytest.fixture(scope='class')
+def vlan_mask(vlan, addr_family):
+    '''Return the correct network mask for the macvlan interface'''
+    return vlan.get_address(addr_family)[1]
```

Comment:
First, this isn't the mask - its the prefix.

And I don't think you should add this fixture.

`vlan` and `vlan_set` have been the predominant way to get macvlan(s). Your change to `get_address` breaks *all* code that uses either of these two fixtures directly. Realise that `vlan_addr` is just a *convenience* fixture because of the sheer number of tests that only care about having a second IP address.

But we have a lot of tests that ask for 5 more, for example, and this change will break them.